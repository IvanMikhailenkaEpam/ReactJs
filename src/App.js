import 'isomorphic-fetch';
import 'regenerator-runtime/runtime';
import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader';
import { Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';

import './App.css';
import Footer from './components/footer/Footer';
import PageNotFoundComponent from './pages/PageNotFoundComponent';
import ErrorBoundary from './components/error-boundary/ErrorBoundaty';
import MovieSearch from './pages/MovieSearch';
import MovieSelected from './pages/MovieSelected';


const App = ({
  Router, location, context, store,
}) => (
  <Provider store={store}>
    <ErrorBoundary>
      <Router location={location} context={context}>
        <Switch>
          <Route exact path="/" component={MovieSearch} />
          <Route path="/search" component={MovieSearch} />
          <Route path="/movie/:id" component={MovieSelected} />
          <Route component={PageNotFoundComponent} />
        </Switch>
        <Footer />

      </Router>
    </ErrorBoundary>
  </Provider>
);

App.propTypes = {
  Router: PropTypes.func.isRequired,
  location: PropTypes.string,
  context: PropTypes.shape({
    url: PropTypes.string,
  }),
  store: PropTypes.shape({
    dispatch: PropTypes.func.isRequired,
    getState: PropTypes.func.isRequired,
  }).isRequired,
};
App.defaultProps = {
  location: null,
  context: null,
};

export default hot(module)(App);
