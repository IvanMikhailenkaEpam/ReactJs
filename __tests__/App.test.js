import React from 'react';
import { shallow } from 'enzyme';
import App from "../src/App";

describe('App', () => {
  it('renders App correctly', () => {
    const component = shallow(<App />);

    expect(component).toMatchSnapshot();
  });
});
