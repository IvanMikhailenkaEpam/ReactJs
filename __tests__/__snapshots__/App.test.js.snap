// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App correctly 1`] = `
<Fragment>
  <BrowserRouter>
    <ErrorBoundary>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/search"
      />
      <Route
        component={[Function]}
        path="/movie/:id"
      />
    </ErrorBoundary>
  </BrowserRouter>
  <Footer />
</Fragment>
`;
